lib/nerves_hub/fwup.ex

```diff
@@ -22,9 +22,8 @@ defmodule NervesHub.Fwup do
   Decode and parse metadata from a FWUP file.
   """
   @spec metadata(String.t()) ::
-          {:ok, map()}
-          | {:error, :invalid_fwup_file}
-          | {:error, :invalid_metadata}
+          {:ok, struct()}
```

Comment:
```suggestion
          {:ok, Metadata.t()}
```
Can you make the `t()` work here so Dialyzer can know about the fields?